util/network/src/error.rs

```diff
@@ -224,27 +224,21 @@ fn test_errors() {
 fn test_io_errors() {
 	use libc::{EMFILE, ENFILE};
 
-	let en_file_error = io::Error::from_raw_os_error(ENFILE);
-
-	match *<Error as From<io::Error>>::from(en_file_error).kind() {
-		ErrorKind::ProcessTooManyFiles => {},
-		_ => panic!("Unexpected error"),
-	}
-
-
-	let em_file_error = io::Error::from_raw_os_error(EMFILE);
-
-	match *<Error as From<io::Error>>::from(em_file_error).kind() {
-		ErrorKind::SystemTooManyFiles => {},
-		_ => panic!("Unexpected error"),
-	}
-
-
-	let usual_error = io::Error::from_raw_os_error(0);
-
-	match *<Error as From<io::Error>>::from(usual_error).kind() {
-		ErrorKind::Io(_) => {},
-		_ => panic!("Unexpected error"),
-	}
-
+	assert_matches!(
+		<Error as From<io::Error>>::from(
```

Comment:
also, is there any reason to do, `Error as From<...>>::from` instead of `Error::from(...)` over here? :)

Reply:
Not strictly, it does work without. I started the test by copy-pasting the tests from above and kept the same explicit statement. I suspect that it was being done there (and why I left it here) to ensure - and make explicit to other people reading the code later - which code part is expected to run in for the test case.